azure-iot-hub/azure/iot/hub/iothub_amqp_client.py

```diff
@@ -100,3 +74,57 @@ def send_message_to_device(self, device_id, message, app_props):
         results = self.amqp_client.send_all_messages(close_on_done=False)
         if uamqp.constants.MessageState.SendFailed in results:
             raise Exception("C2D message send failure")
+
+
+class IoTHubAmqpClientSharedAccessKeyAuth(IoTHubAmqpClientBase):
+    def __init__(self, hostname, shared_access_key_name, shared_access_key):
+        self.endpoint = self._build_amqp_endpoint(
+            hostname, shared_access_key_name, shared_access_key
+        )
+        operation = "/messages/devicebound"
+        target = "amqps://" + self.endpoint + operation
+        self.amqp_client = uamqp.SendClient(target)
+
+    def _generate_auth_token(self, uri, sas_name, sas_value):
```

Comment:
Could this not just leverage the sastoken object from `azure.iot.hub.sastoken`?

Looking at it now, it seems like the logic is the same. If you leveraged it, it would reduce duplication of code + make the tests easier to write.